### PR TITLE
refactor(ingest): move Airflow into `datahub_provider` module

### PR DESCRIPTION
**Pull request by hsheth2**
Airflow users expect stuff to be in `datahub_provider.{hooks,operators,lineage}`. 
Maintains backwards compatibility by re-exporting from the `datahub.integrations.airflow` files.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by hsheth2** [comment]
@sunkickr makes sense - just made the change

File: metadata-ingestion/tests/unit/test_airflow.py [hunk 1]
```diff
@@ -19,9 +19,9 @@
```
Comment (sunkickr) [inline metadata-ingestion/tests/unit/test_airflow.py:19]: You may want to store tests for operators, hooks, and the lineagebackend in datahub_provider like done in the sample repo: https://github.com/astronomer/airflow-provider-sample/tree/main/tests. This way modules from datahub_provider won't need to be imported into datahub.

**Review by sunkickr — COMMENTED** [review]
Just like my comment on tests above, example airflow dags can also be included in the datahub_provider like in the sample_repo here: https://github.com/astronomer/airflow-provider-sample/tree/main/sample_provider/example_dags. This way all airflow stuff can live in the provider and the registry will be able to build a page for each example dag in the example_dags directory like here: https://registry.astronomer-stage.io/providers/google/example-dags/example-automl-nl-text-classification

File: metadata-ingestion/tests/unit/test_airflow.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/tests/unit/test_airflow.py:19]: Not sure I'm following the suggestion - the tests in both the sample repo and this one are stored separate from the main source directories.

The `datahub` imports of `datahub_provider` are primarily for backwards compatibility, so I'm pretty ok with leaving those in the code.

**Comment by sunkickr** [comment]
@hsheth2  - Maybe we could just duplicate the example dags repo into the provider package so they are easily discoverable in both the datahub repo and the registry.  Example dags located in the `datahub_provider/example_dags/` directory will be searchable in the registry. Also we are planning lots of features for example dags so provider packages that have them will get more exposure by having more searchable/interactive content. Right now if a user is looking at a datahub module in the registry, example dags that use that module will appear with it if they are included in the package.

File: metadata-ingestion/tests/unit/test_airflow.py (hunk 1 above)
Comment (sunkickr) [reply metadata-ingestion/tests/unit/test_airflow.py:19]: That makes sense was just suggesting because module tests are usually packaged with modules.

**Comment by sunkickr** [comment]
Everything looks good to me as far as structure goes.

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**